app/controllers/products_controller.rb

```diff
@@ -40,9 +40,9 @@ def update
 
   def destroy
     @shopify_product = ShopifyAPI::Product.find(session: @session, id: @product.shopify_id)
-    if @product.destroy && @shopify_product.delete
-      redirect_to products_path, notice: 'The product was successfully destroyed.'
-    end
+    return unless @product.destroy && @shopify_product.delete
```

Comment:
When you are converting if to unless you need to change logical operators to opposites.
if a < b && c < d == unless a < b || c < d
https://brilliant.org/wiki/de-morgans-laws/#:~:text=De%20Morgan's%20Laws%20describe%20how,disjunctions%20of%20propositions%20through%20negation.